src/main/java/org/jabref/logic/xmp/DublinCoreExtractor.java

```diff
@@ -0,0 +1,387 @@
+package org.jabref.logic.xmp;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.Map.Entry;
+
+import org.jabref.logic.TypedBibEntry;
+import org.jabref.model.database.BibDatabaseMode;
+import org.jabref.model.entry.Author;
+import org.jabref.model.entry.AuthorList;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.entry.Month;
+
+import org.apache.xmpbox.DateConverter;
+import org.apache.xmpbox.schema.DublinCoreSchema;
+
+public class DublinCoreExtractor {
+
+    private final DublinCoreSchema dcSchema;
+    private final XMPPreferences xmpPreferences;
+
+    private final BibEntry bibEntry;
+
+    public DublinCoreExtractor(DublinCoreSchema dcSchema, XMPPreferences xmpPreferences, BibEntry resolvedEntry) {
+        this.dcSchema = dcSchema;
+        this.xmpPreferences = xmpPreferences;
+
+        this.bibEntry = resolvedEntry;
+    }
+
+    /**
+     * Editor in BibTex - Contributor in DublinCore
+     *
+     * @param bibEntry The BibEntry object, which is filled during metadata extraction.
+     * @param dcSchema Metadata in DublinCore format.
+     */
+    private void extractEditor() {
+        List<String> contributors = dcSchema.getContributors();
+        if ((contributors != null) && !contributors.isEmpty()) {
+            bibEntry.setField(FieldName.EDITOR, String.join(" and ", contributors));
+        }
+    }
+
+    /**
+     * Author in BibTex - Creator in DublinCore
+     *
+     * @param bibEntry The BibEntry object, which is filled during metadata extraction.
+     * @param dcSchema Metadata in DublinCore format.
+     */
+    private void extractAuthor() {
+        List<String> creators = dcSchema.getCreators();
+        if ((creators != null) && !creators.isEmpty()) {
+            bibEntry.setField(FieldName.AUTHOR, String.join(" and ", creators));
+        }
+    }
+
+    /**
+     * Year + Month in BibTex - Date in DublinCore is a combination of year and month information
+     *
+     * @param bibEntry The BibEntry object, which is filled during metadata extraction.
+     * @param dcSchema Metadata in DublinCore format.
+     */
+    private void extractYearAndMonth() {
+        List<String> dates = dcSchema.getUnqualifiedSequenceValueList("date");
+        if ((dates != null) && !dates.isEmpty()) {
+            String date = dates.get(0).trim();
+            Calendar c = null;
```

Comment:
Please stick to the java 8 new date and time api: Use a DateTime or just a DateFormatter 
DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
LocalDateTime dateTime = LocalDateTime.parse(str, formatter);